### DISCUSSION TITLE
I need to change the image that is being edited

### DISCUSSION THREAD

**Discussion by wfettich** [guoyingtao/Mantis#378] [Q&A]
Hi, 
I need to apply a filter on the mage while it is being edited. I managed to do a CustomViewController that also contains a slider that sets a certain filter value. However I can't seem to find how to change the image that is being edited. Could someone help me with some pointers or an example ? 

I create the editor using the helper functions: Mantis.cropViewController .. where I pass the image to edit. However I cannot seem to be able to change it after the editor was created. 

Thank you.

**Comment by guoyingtao** [comment]
@wfettich 
Currently Mantis does not support the feature you mentioned. 
Is the demo below is something you want? (I changed some code for a test)
If you can provide more details of your requirement, I will consider to add this support in a future version.

https://github.com/guoyingtao/Mantis/assets/26723384/33a6ce19-33c5-419f-ab12-4d6e491e88ea

**Comment by wfettich** [comment]
Yes, that's it.
But I have my own slider that applies my filter to that image.

**Comment by wfettich** [comment]
Do you think you could share the code for that demo ?

**Comment by wfettich** [comment]
Basically I have this, and when I move that silder I want to apply my
filter to the image.

**Comment by guoyingtao** [comment]
If you just want to have a test, you can add the code below to CropViewController, and remove `private` for property  imageView in ImageContainer

```Swift
    public func setImage(_ image: UIImage) {
        cropView.image = image
        ((cropView as! CropView).imageContainer as! ImageContainer).imageView.image = image
    }

```

The code is for test only. I need a better API to support this feature if needed.

**Comment by guoyingtao** [comment]
Now you can have a test on this PR #387
